wagtail/models/sites.py

```diff
@@ -215,55 +286,79 @@ def get_site_root_paths():
         - `root_path` - The internal URL path of the site's home page (for example '/home/')
         - `root_url` - The scheme/domain name of the site (for example 'https://www.example.com/')
         - `language_code` - The language code of the site (for example 'en')
-        """
-        result = cache.get(
```

Comment:
Issue: This will now always be recalculated every thread, rather than being able to rely on a shared cache (even if the cache is outside memory). It would be nice to maintain the existing cache as a form of tiered cache:

1. `asgiref.Local`
2. `django.core.cache`
3. Miss - recalculate

Reply:
Personally, I think that's a lot of extra code complexity and tests for not a lot of gain. But, I'm happy to be convinced otherwise. @gasman Do you have any thoughts on this?